clients/fides-js/src/components/tcf/TcfOverlay.tsx

```diff
@@ -401,12 +485,19 @@ export const TcfOverlay = ({
               return (
                 <TcfConsentButtons
                   experience={experience}
-                  onSave={onSave}
+                  onAcceptAll={() => {
+                    handleAcceptAll();
+                    onClose();
+                  }}
+                  onRejectAll={() => {
+                    handleRejectAll();
+                    onClose();
+                  }}
                   renderFirstButton={() => (
                     <Button
                       buttonType={ButtonType.SECONDARY}
                       label={i18n.t("exp.save_button_label")}
-                      onClick={() => onSave(ConsentMethod.SAVE, draftIds!)}
+                      onClick={() => onSave(ConsentMethod.SAVE, draftIds)}
```

Comment:
what did the `!` actually do before?

Reply:
Its purpose is to tell the Typescript validator to ignore the fact that it might be null/undefined whereas (in this case) the `onSave` method expects it to have a value. Usually reserved for something where the developer is 100% sure that value will be set by the time it hits this chunk of code, I find its usually abused and prone to unexpected bugs so I prefer cleaning it up wherever possible. In this case I set the value explicitly on line 174, giving the `useState` a default.